src/pocketmine/Player.php

```diff
@@ -2368,6 +2368,10 @@ public function handleDataPacket(DataPacket $packet){
 				$this->iusername = strtolower($this->username);
 				$this->protocol = $packet->protocol;
 
+				if($this->server->getConfigBoolean("online-mode", false) && $packet->identityPublicKey === null){
+					$this->kick("disconnectionScreen.notAuthenticated", false);
```

Comment:
I think this needs to be close() and not kick(). See below disconnection screens.